lib/archiver/archiver.rb

```diff
@@ -81,6 +98,20 @@ def self.external_archive_url?(url)
     allowed_hosts.any? {|base| url.starts_with?(base)}
   end
 
+  def self.get_file_from_uri(uri)
+    # HTTP GET the file if its URI belongs to a public host.
+    if public_archive_url?(uri)
```

Comment:
I don’t think we ever execute this branch of logic in the current implementation. Is there an important use case you’re anticipating here? Otherwise, let’s drop it.

Reply:
I was just making sure this method works as advertised in case there's some other use for it in the future. No idea what that use might be, but if this is supposed to be a public API, I'd leave it in for completeness; if not I'll drop it.

Reply:
I think I might rename this method to something more like `get_archived_file()`, which wouldn’t imply this would ever go out to the internet, and you might not worry so much about needing to have something like this for completeness’s sake. That would also make it a clearer analog to `Archiver.archive()`.

Reply:
dispensed with this.